src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -3283,6 +3283,174 @@ def groupby_apply(
             )
         )
 
+    def _fill_null_values_for_groupby_first_last(
```

Comment:
This looks like most of what we need for groupby.fillna(). Should we implement that in a separate PR first?

Reply:
I also thought about just adding it to this PR, but I thought it would get too bloated.